hugr-model/src/v0/binary/read.rs

```diff
@@ -245,22 +211,35 @@ fn read_region<'a>(
         hugr_capnp::RegionKind::Module => model::RegionKind::Module,
     };
 
-    let sources = read_list!(bump, reader, get_sources, read_link_ref);
-    let targets = read_list!(bump, reader, get_targets, read_link_ref);
+    let sources = read_scalar_list!(bump, reader, get_sources, model::LinkIndex);
+    let targets = read_scalar_list!(bump, reader, get_targets, model::LinkIndex);
     let children = read_scalar_list!(bump, reader, get_children, model::NodeId);
     let meta = read_list!(bump, reader, get_meta, read_meta_item);
     let signature = reader.get_signature().checked_sub(1).map(model::TermId);
 
+    let scope = if reader.has_scope() {
```

Comment:
The `if` checks if the field is set to see if the region is open (field absent) or closed (field present). I'm not entirely sure if `get_scope` on an absent scope would be an error or return a default value, but either would not be desirable: An error would be undesirable since we want to distinguish between an open region and a corrupted file. A default value would make every region closed.